02-express-tutorial/answers/15-router-controller.js

```diff
@@ -0,0 +1,61 @@
+let { people } = require('../data');
+
+const getPeople = (req, res) => {
+  res.status(200).json({ success: true, data: people });
+};
+
+const createPerson = (req, res) => {
+  const { name } = req.body;
+  if (!name) {
+    return res
+      .status(400)
+      .json({ success: false, msg: 'please provide a name' });
+  }
+  res.status(201).json({ success: true, person: name });
+};
+const createPersonPostman = (req, res) => {
+  const { name } = req.body;
+  if (!name) {
+    return res
+      .status(400)
+      .json({ success: false, msg: 'please provide a name' });
+  }
+  res.status(201).json({ success: true, data: [...people, name] });
+};
+
+const updatePerson = (req, res) => {
+  const { id } = req.params;
+  const { name } = req.body;
+  const person = people.find((person) => person.id === Number(id));
+  if (!person) {
+    return res
+      .status(404)
+      .json({ success: false, msg: `person not found with this ${id}` });
+  }
+  const newPeople = people.map((person) => {
+    if (person.id === Number(id)) {
```

Comment:
We call the Number constructor with `id` two times. Can we refactor to avoid that?

Reply:
Yes, we can create a variable and store the parsed Id value like so: 
 const parsedId  = Number(id); 
and out if statement in this case will look like this: 
 if (person.id === parsedId) {
//block of code
 }